### ISSUE TITLE
Change document title from "Storybook" to "Cloud Four Patterns"

### ISSUE THREAD

**Issue by spaceninja** [cloudfour/cloudfour.com-patterns#536]
Unfortunately, there is currently no mechanism to do so. This used to be supported via the now-deprecated [addon-options](https://github.com/storybooks/storybook/tree/master/addons/options).

There's a lengthy, ongoing, open issue, and several other closed issues regarding this topic. I tried every solution in https://github.com/storybookjs/storybook/issues/866 but none of them work — there's a bit of React logic happening that immediately resets the document title to "Storybook" after a second.

For now, this ticket is not actionable, but we should keep an eye on that ticket and apply a solution when one exists.

**Comment by tylersticka** [comment]
We have updated to Storybook 6.5.x, so this should not be blocked anymore.

**Comment by tylersticka** [comment]
I think there was a misunderstanding in that issue of what the request was. I still think this isn't possible without hackery until https://github.com/storybookjs/storybook/issues/14726 is resolved.

Marking as blocked again.

**Comment by spaceninja** [comment]
Closing this. Storybook does not intend to support it, and the only route left is a hack we don't want.

## Upstream has declined it

This was marked blocked on [storybookjs/storybook#14726](https://github.com/storybookjs/storybook/issues/14726), which is now **closed as not planned**. [#6339](https://github.com/storybookjs/storybook/issues/6339), the other ticket covering the same request, is closed too. Waiting for upstream is no longer a strategy, so the `✋ blocked` label no longer describes anything.

There is also no configuration hook to have missed. On Storybook 10.5.10 there is no `documentTitle`, `pageTitle` or `titleTemplate` anywhere in the published types, and `brandTitle` — which we already set in `.storybook/cloud-four-theme.js` — only feeds the sidebar logo and its link.

The title is hard-coded in the manager runtime:

```js
// `${title} - ${name} ⋅ Storybook`, or plain "Storybook" with no entry
getDescription = (item) => { /* ... */ }
```

and applied through `react-helmet-async`, which is the "bit of React logic happening that immediately resets the document title" described in the original report six years ago. It is still exactly that.

## What has changed is that the gap is now much smaller

The complaint this was opened about — that the tab just says "Storybook" — is now only true of the shell before a story loads:

| State | Tab title |
| --- | --- |
| Static shell, before a story is selected | `storybook - Storybook` |
| Any story selected | `Components / Button - Default ⋅ Storybook` |
| Any docs page | `Components / Cloud Cover - Scene ⋅ Storybook` |

So the tab is already informative, and it already tells you which story you are looking at, which is more than we were asking for. What is left is a four-word suffix reading `⋅ Storybook` rather than `⋅ Cloud Four Patterns`.

## Why not hack it

Two options exist and both are worse than the suffix:

- A manager addon that re-sets `document.title` on `STORY_CHANGED` races Helmet, because Helmet writes the title from its own commit phase. This is the approach that "works for about half a second" in the upstream threads.
- Patching the `document.title` setter to rewrite the suffix is deterministic rather than racy, but it monkey-patches a DOM accessor and pattern-matches on a hard-coded Storybook string. It would break silently on any Storybook release that reworded that string, and the failure mode is a wrong browser tab that nobody notices for months.

Neither is worth carrying in `.storybook/` to change a suffix on a title that already names the story. Closing as won't do; if Storybook ever adds a real option, this is cheap to revisit.
